storage_service/locations/api/beta/remple/openapi.py

```diff
@@ -0,0 +1,1573 @@
+        read_schema['properties'] = read_schema_properties
+        if required_fields:
+            read_schema['required'] = required_fields
+        return read_schema_name, read_schema
+
+    def get_api_version_slug(self):
+        try:
+            return self._get_api_version_slug(self.api_version)
+        except ValueError:
+            return self.api_version
+
+    @staticmethod
+    def _get_api_version_slug(version):
+        """Given a version number like 'X.Y.Z', return a slug representation of
+        it. E.g.,
+
+            >>> get_api_version_slug('3.0.0')
+            ... 'v3'
+            >>> get_api_version_slug('3.0.1')
+            ... 'v3_0_1'
+            >>> get_api_version_slug('3.0')
+            ... 'v3'
+            >>> get_api_version_slug('3.9')
+            ... 'v3_9'
+        """
+        parts = version.strip().split('.')
+        new_parts = []
+        for index, part in enumerate(parts):
+            part_int = int(part)
+            if part_int:
+                new_parts.append(part)
+            else:
+                parts_to_right = parts[index + 1:]
+                non_empty_ptr = [p for p in parts_to_right if int(p)]
+                if non_empty_ptr:
+                    new_parts.append(part)
+        return 'v{}'.format('_'.join(new_parts))
+
+
+def single_reference_field_dict(**kwargs):
+    """Return an OpenAPI OrderedDict for a Django ForeingKey or OneToOneRel.
+    """
+    if isinstance(kwargs['field'], OneToOneRel):
+        return (kwargs['field'].get_accessor_name(),
+                OrderedDict([('type', 'string'), ('format', 'uri')]))
+    return (kwargs['field_name'],
+            OrderedDict([('type', 'string'), ('format', 'uri')]))
+
+
+def multi_reference_field_dict(**kwargs):
+    """Return an OpenAPI OrderedDict for a Django ManyToManyField,
+    ManyToManyRel, or ManyToOneRel.
+    """
+    field_name = kwargs.get('field_name')
+    if isinstance(kwargs['field'], ManyToOneRel):
+        field_name = kwargs['field'].get_accessor_name()
+    return field_name, OrderedDict([
+        ('type', 'array'),
+        ('items', OrderedDict([
+            ('type', 'string'), ('format', 'uri')]))])
+
+
+def scalar_field_dict(**kwargs):
+    """Return an OpenAPI OrderedDict for a Django scalar field, e.g., a string
+    or an int.
+    """
+    field = kwargs.get('field')
+    openapi_type = django_field2openapi_type.get(
+        type(field))
+    field_dict = OrderedDict([('type', openapi_type)])
+    openapi_format = django_field2openapi_format.get(
+        type(field))
+    if openapi_format:
+        field_dict['format'] = openapi_format
+    return kwargs['field_name'], field_dict
+
+
+def get_choices(field):
+    choices = getattr(field, 'choices', None)
+    if choices:
+        return [c[0] for c in choices]
+    return None
+
+
+def get_default(field):
+    return getattr(field, 'default', NOT_PROVIDED)
+
+
+def get_required(field, default):
```

Comment:
Fix function name since it returns a boolean indicating whether a field is required and whether it is null(able).